config/xml_schemas/env_batch.xsd

```diff
@@ -5,7 +5,7 @@
 
   <!-- attributes -->
   <xs:attribute name="id" type="xs:NCName"/>
-  <xs:attribute name="default" type="xs:NCName"/>
```

Comment:
This is the wrong field in the wrong file. The correct one is [here:](https://github.com/ESMCI/cime/blob/maint-5.6/config/xml_schemas/config_batch.xsd#L8)

Reply:
That link is a little deceptive because I'm pointing to our copy of config_batch.xsd, if you check yours you will see the type="xs:NCName"

Reply:
Apologies, I do not understand - the highlighted text line 8  in file env_batch.xsd in this reply is where I changed xs:NCName to "xs:string" as you directed in an earlier comment.  Line 8 of config_batch.xsd already has "xs:string".
Are you revising the suggestion for me to change revert a change and change something else?
I can try use a different queue name for testing (it is someone else's node but the name is smbgen so none numeric).

Reply:
Sorry - I am revising again to say that the correct line to change is 
around 174 of config_batch.xsd
```
  <xs:extension base="xs:NCName">      
```

Reply:
OKAY. Progress. Now we have compilation errors - looks like I missed NETCDF library environment variable somewhere.
Back to the "porting" guidance I think.  But  this is great progress. Thanks!
```
[earmgr@login2.arc4 my_cesm_sandbox]$ module li
Currently Loaded Modulefiles:
  1) licenses              3) intel/19.0.4          5) user                  7) hdf5/1.8.21
  2) sge                   4) intelmpi/2019.4.243   6) mkl/2019.0            8) netcdf/4.6.3
```


```
[earmgr@login2.arc4 my_cesm_sandbox]$ ${CIMEROOT}/scripts/create_test SMS.f19_g17.X
Testnames: ['SMS.f19_g17.X.arc4_intel']
No project info available
Creating test directory /nobackup/earmgr/cesm213/my_cesm_sandbox/cesm/SMS.f19_g17.X.arc4_intel.20210616_175713_pixyta
RUNNING TESTS:
  SMS.f19_g17.X.arc4_intel
Starting CREATE_NEWCASE for test SMS.f19_g17.X.arc4_intel with 1 procs
Finished CREATE_NEWCASE for test SMS.f19_g17.X.arc4_intel in 1.209489 seconds (PASS)
Starting XML for test SMS.f19_g17.X.arc4_intel with 1 procs
Finished XML for test SMS.f19_g17.X.arc4_intel in 0.231847 seconds (PASS)
Starting SETUP for test SMS.f19_g17.X.arc4_intel with 1 procs
Finished SETUP for test SMS.f19_g17.X.arc4_intel in 1.381910 seconds (PASS)
Starting SHAREDLIB_BUILD for test SMS.f19_g17.X.arc4_intel with 1 procs
Finished SHAREDLIB_BUILD for test SMS.f19_g17.X.arc4_intel in 97.174612 seconds (PASS)
Starting MODEL_BUILD for test SMS.f19_g17.X.arc4_intel with 4 procs
Finished MODEL_BUILD for test SMS.f19_g17.X.arc4_intel in 15.807277 seconds (FAIL). [COMPLETED 1 of 1]
    Case dir: /nobackup/earmgr/cesm213/my_cesm_sandbox/cesm/SMS.f19_g17.X.arc4_intel.20210616_175713_pixyta
    Errors were:
        Building test for SMS in directory /nobackup/earmgr/cesm213/my_cesm_sandbox/cesm/SMS.f19_g17.X.arc4_intel.20210616_175713_pixyta
        /nobackup/earmgr/cesm213/my_cesm_sandbox/cime/src/share/util/shr_mct_mod.F90:107: undefined reference to `netcdf_mp_nf90_open_'

```
(lots of unresolved netcdf remarks)

END OF MSG for brevity

Reply:
To check the modules you are using for the build go to the case directory and run
```
source .env_mach_specific.sh
```  
(or .csh if that's your shell) then do the module list command.

Reply:
Different machines define modules in different ways.  In some cases you will need to modify the config_compilers.xml file for your system, you should find plenty of examples therein.